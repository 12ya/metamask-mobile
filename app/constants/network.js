import { NetworkType } from '@metamask/controller-utils';

export const MAINNET = 'mainnet';
export const HOMESTEAD = 'homestead';
export const GOERLI = 'goerli';
export const SEPOLIA = 'sepolia';
export const LINEA_GOERLI = 'linea-goerli';
export const LINEA_MAINNET = 'linea-mainnet';
export const RPC = NetworkType.rpc;
export const NO_RPC_BLOCK_EXPLORER = 'NO_BLOCK_EXPLORER';
export const PRIVATENETWORK = 'PRIVATENETWORK';
export const DEFAULT_MAINNET_CUSTOM_NAME = 'Ethereum Main Custom';
export const IPFS_DEFAULT_GATEWAY_URL = 'https://cloudflare-ipfs.com/ipfs/';

/**
 * @enum {string}
 */
export const NETWORKS_CHAIN_ID = {
  MAINNET: '1',
  OPTIMISM: '10',
  BSC: '56',
  POLYGON: '137',
  FANTOM: '250',
  BASE: '8453',
  ARBITRUM: '42161',
  AVAXCCHAIN: '43114',
  CELO: '42220',
  HARMONY: '1666600000',
  SEPOLIA: '11155111',
  LINEA_GOERLI: '59140',
  GOERLI: '5',
  LINEA_MAINNET: '59144',
  ZKSYNC_ERA: '324',
};

/**
 * @enum {string}
 */
export const NETWORKS_CHAIN_ID_WITH_SVG = {
  MAINNET: '1',
  OPTIMISM: '10',
  BSC: '56',
  POLYGON: '137',
  FANTOM: '250',
  BASE: '8453',
  ARBITRUM: '42161',
  AVAXCCHAIN: '43114',
  CELO: '42220',
  LINEA_MAINNET: '59144',
  ZKSYNC_ERA: '324',
  ACALA_NETWORK: '787',
  ARBITRUM_NOVA: '42170',
  ARBITRUM_ONE: '42161',
  ASTAR: '592',
  CANTO: '7700',
  CONFLUX: '1030',
  CORE_MAINNET: '1116',
  CRONOS: '25',
  DEXALOT: '432204',
  ETH_CLASSIC: '61',
  EVMOS: '9001',
  FLARE: '14',
  FUSE: '122',
  GNOSIS: '100',
  HAQQ: '11235',
  KAVA: '2222',
  KCC: '321',
  KLAYTN: '8217',
  KROMA: '255',
  LIGHT_LINK: '1890',
  MANTA: '169',
  MANTLE: '5000',
  MOONBEAM: '1284',
  MOONRIVER: '1285',
  NEAR_AURORA: '1313161554',
  NEBULA: '1482601649',
  OASYS: '248',
  OKX: '66',
  OP_BNB: '204',
  PGN: '424',
  ZKEVM_POLYGON: '1101',
  PULSE: '369',
  SCROLL: '534352',
  SHARDEUM: '8081',
  SHARDEUM_SPHINX: '8082',
  SHIB: '27',
  SONGBIRD: '19',
  STEP_NETWORK: '1234',
  TAIKO: '167008',
  TENET: '1559',
  VELAS: '106',
  ZORA: '7777777',
};
