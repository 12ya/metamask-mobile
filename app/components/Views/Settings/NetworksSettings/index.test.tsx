import { renderScreen } from '../../../../util/test/renderWithProvider';
import NetworksSettings from './';
import { backgroundState } from '../../../../util/test/initial-root-state';
import { mockNetworkState } from '../../../../util/test/network';
import { RpcEndpointType } from '@metamask/network-controller';

const mockInitialState = {
  settings: {},
  engine: {
    backgroundState: {
      ...backgroundState,
      NetworkController: {
        ...mockNetworkState({
          id: 'mainnet',
          nickname: 'Ethereum Mainnet',
          ticker: 'ETH',
          chainId: '0x1',
          type: RpcEndpointType.Infura,
        }),
      },
    },
  },
};

jest.mock('../../../../store', () => ({
  store: {
    getState: () => mockInitialState,
  },
}));

describe('NetworksSettings', () => {
  it('should render correctly', () => {
    const { toJSON } = renderScreen(
      NetworksSettings,
      { name: 'Network Settings' },
      {
        state: mockInitialState,
      },
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
