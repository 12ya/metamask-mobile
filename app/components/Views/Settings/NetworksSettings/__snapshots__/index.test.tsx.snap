// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworksSettings should render correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "#ffffff",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "elevation": 0,
                  "flex": 1,
                  "shadowColor": "transparent",
                  "shadowOffset": {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              >
                <TouchableOpacity
                  accessible={true}
                  activeOpacity={1}
                  disabled={false}
                  onPress={[Function]}
                  onPressIn={[Function]}
                  onPressOut={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "borderRadius": 8,
                      "height": 32,
                      "justifyContent": "center",
                      "marginHorizontal": 16,
                      "opacity": 1,
                      "width": 32,
                    }
                  }
                  testID="back-arrow-button"
                >
                  <SvgMock
                    color="#141618"
                    height={24}
                    name="ArrowLeft"
                    style={
                      {
                        "height": 24,
                        "width": 24,
                      }
                    }
                    width={24}
                  />
                </TouchableOpacity>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 590,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    {
                      "color": "#0376c9",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  Networks
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            collapsable={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "backgroundColor": "#ffffff",
                            "flex": 1,
                            "marginBottom": 24,
                            "paddingHorizontal": 24,
                            "paddingVertical": 12,
                          }
                        }
                        testID="networks-screen"
                      >
                        <View
                          style={
                            {
                              "alignItems": "center",
                              "borderColor": "#bbc0c5",
                              "borderRadius": 5,
                              "borderWidth": 1,
                              "color": "#141618",
                              "flexDirection": "row",
                              "paddingHorizontal": 10,
                              "paddingVertical": 10,
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              [
                                {
                                  "color": "#141618",
                                  "fontSize": 20,
                                },
                                undefined,
                                {
                                  "fontFamily": "Ionicons",
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                {},
                              ]
                            }
                          >
                            
                          </Text>
                          <TextInput
                            onChangeText={[Function]}
                            placeholder="Search for previously added network"
                            placeholderTextColor="#141618"
                            style={
                              {
                                "color": "#141618",
                                "flex": 1,
                                "fontFamily": "EuclidCircularB-Regular",
                                "fontSize": 14,
                                "fontWeight": "400",
                                "paddingLeft": 10,
                              }
                            }
                            testID="network-search-input"
                            value=""
                          />
                        </View>
                        <RCTScrollView>
                          <View>
                            <Text
                              style={
                                {
                                  "color": "#141618",
                                  "fontFamily": "EuclidCircularB-Bold",
                                  "fontSize": 14,
                                  "fontWeight": "600",
                                  "paddingVertical": 12,
                                }
                              }
                            >
                              Mainnet
                            </Text>
                            <View>
                              <TouchableOpacity
                                onPress={[Function]}
                                style={
                                  {
                                    "alignItems": "center",
                                    "flex": 1,
                                    "flexDirection": "row",
                                    "paddingVertical": 12,
                                  }
                                }
                              >
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "flex": 0,
                                      "flexDirection": "row",
                                    }
                                  }
                                >
                                  <Image
                                    source={
                                      {
                                        "uri": "MockImage",
                                      }
                                    }
                                    style={
                                      {
                                        "borderRadius": 10,
                                        "height": 20,
                                        "marginRight": 16,
                                        "marginTop": 2,
                                        "width": 20,
                                      }
                                    }
                                  />
                                  <View>
                                    <Text
                                      style={
                                        {
                                          "color": "#141618",
                                          "fontFamily": "EuclidCircularB-Regular",
                                          "fontSize": 16,
                                          "fontWeight": "400",
                                        }
                                      }
                                    >
                                      Ethereum Main Network
                                    </Text>
                                  </View>
                                </View>
                                <Text
                                  allowFontScaling={false}
                                  style={
                                    [
                                      {
                                        "color": "#141618",
                                        "fontSize": 20,
                                      },
                                      {
                                        "marginLeft": 8,
                                      },
                                      {
                                        "fontFamily": "FontAwesome",
                                        "fontStyle": "normal",
                                        "fontWeight": "normal",
                                      },
                                      {},
                                    ]
                                  }
                                >
                                  
                                </Text>
                              </TouchableOpacity>
                            </View>
                            <View>
                              <TouchableOpacity
                                onPress={[Function]}
                                style={
                                  {
                                    "alignItems": "center",
                                    "flex": 1,
                                    "flexDirection": "row",
                                    "paddingVertical": 12,
                                  }
                                }
                              >
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "flex": 0,
                                      "flexDirection": "row",
                                    }
                                  }
                                >
                                  <Image
                                    source={
                                      {
                                        "uri": "MockImage",
                                      }
                                    }
                                    style={
                                      {
                                        "borderRadius": 10,
                                        "height": 20,
                                        "marginRight": 16,
                                        "marginTop": 2,
                                        "width": 20,
                                      }
                                    }
                                  />
                                  <View>
                                    <Text
                                      style={
                                        {
                                          "color": "#141618",
                                          "fontFamily": "EuclidCircularB-Regular",
                                          "fontSize": 16,
                                          "fontWeight": "400",
                                        }
                                      }
                                    >
                                      Linea Main Network
                                    </Text>
                                  </View>
                                </View>
                                <Text
                                  allowFontScaling={false}
                                  style={
                                    [
                                      {
                                        "color": "#141618",
                                        "fontSize": 20,
                                      },
                                      {
                                        "marginLeft": 8,
                                      },
                                      {
                                        "fontFamily": "FontAwesome",
                                        "fontStyle": "normal",
                                        "fontWeight": "normal",
                                      },
                                      {},
                                    ]
                                  }
                                >
                                  
                                </Text>
                              </TouchableOpacity>
                            </View>
                            <Text
                              style={
                                {
                                  "color": "#141618",
                                  "fontFamily": "EuclidCircularB-Bold",
                                  "fontSize": 14,
                                  "fontWeight": "600",
                                  "paddingVertical": 12,
                                }
                              }
                            >
                              Test networks
                            </Text>
                            <View>
                              <TouchableOpacity
                                onLongPress={[Function]}
                                onPress={[Function]}
                              >
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "flex": 1,
                                      "flexDirection": "row",
                                      "paddingVertical": 12,
                                    }
                                  }
                                >
                                  <Image
                                    source={
                                      {
                                        "uri": "MockImage",
                                      }
                                    }
                                    style={
                                      {
                                        "borderRadius": 10,
                                        "height": 20,
                                        "marginRight": 16,
                                        "marginTop": 2,
                                        "width": 20,
                                      }
                                    }
                                  />
                                  <Text
                                    style={
                                      {
                                        "color": "#141618",
                                        "fontFamily": "EuclidCircularB-Regular",
                                        "fontSize": 16,
                                        "fontWeight": "400",
                                      }
                                    }
                                  >
                                    Sepolia
                                  </Text>
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      [
                                        {
                                          "color": "#141618",
                                          "fontSize": 20,
                                        },
                                        {
                                          "marginLeft": 8,
                                        },
                                        {
                                          "fontFamily": "FontAwesome",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        {},
                                      ]
                                    }
                                  >
                                    
                                  </Text>
                                </View>
                              </TouchableOpacity>
                            </View>
                            <View>
                              <TouchableOpacity
                                onLongPress={[Function]}
                                onPress={[Function]}
                              >
                                <View
                                  style={
                                    {
                                      "alignItems": "center",
                                      "flex": 1,
                                      "flexDirection": "row",
                                      "paddingVertical": 12,
                                    }
                                  }
                                >
                                  <Image
                                    source={
                                      {
                                        "uri": "MockImage",
                                      }
                                    }
                                    style={
                                      {
                                        "borderRadius": 10,
                                        "height": 20,
                                        "marginRight": 16,
                                        "marginTop": 2,
                                        "width": 20,
                                      }
                                    }
                                  />
                                  <Text
                                    style={
                                      {
                                        "color": "#141618",
                                        "fontFamily": "EuclidCircularB-Regular",
                                        "fontSize": 16,
                                        "fontWeight": "400",
                                      }
                                    }
                                  >
                                    Linea Sepolia
                                  </Text>
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      [
                                        {
                                          "color": "#141618",
                                          "fontSize": 20,
                                        },
                                        {
                                          "marginLeft": 8,
                                        },
                                        {
                                          "fontFamily": "FontAwesome",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        {},
                                      ]
                                    }
                                  >
                                    
                                  </Text>
                                </View>
                              </TouchableOpacity>
                            </View>
                          </View>
                        </RCTScrollView>
                        <TouchableOpacity
                          accessibilityRole="button"
                          accessible={true}
                          activeOpacity={0.2}
                          onPress={[Function]}
                          style={
                            [
                              [
                                {
                                  "borderRadius": 100,
                                  "justifyContent": "center",
                                  "padding": 15,
                                },
                                {
                                  "backgroundColor": "#0376c9",
                                  "minHeight": 50,
                                },
                                undefined,
                              ],
                              null,
                            ]
                          }
                          testID="add-network-button"
                        >
                          <Text
                            style={
                              [
                                {
                                  "color": "#007aff",
                                  "fontSize": 17,
                                  "fontWeight": "500",
                                  "textAlign": "center",
                                },
                                null,
                                [
                                  {
                                    "fontFamily": "EuclidCircularB-Bold",
                                    "fontSize": 14,
                                    "fontWeight": "600",
                                    "textAlign": "center",
                                  },
                                  {
                                    "color": "#ffffff",
                                  },
                                  undefined,
                                ],
                                null,
                              ]
                            }
                          >
                            Add Network
                          </Text>
                        </TouchableOpacity>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
